### ISSUE TITLE
Upstream mongoengine support of Django has ended

### ISSUE THREAD

**Issue by dsnellgrove** [crits/crits#521]
The mongoengine maintainers are no longer supporting django and have removed support in their most recent release https://github.com/MongoEngine/mongoengine/issues/958. If CRITs wants to continue using mongoengine then support may have to be provided to mongoengine. This will be needed to ever move to pymongo 3+.

**Comment by mgoffin** [comment]
I think we'll need this because we moved authentication storage into MongoDB? Definitely need to be explored. Thanks!

**Comment by ckane** [comment]
Is there some other MVC framework that mongoengine is preferring instead?

**Comment by dsnellgrove** [comment]
They seem to have pushed framework-specific code out of the core project:
https://github.com/MongoEngine/eve-mongoengine
https://github.com/MongoEngine/flask-mongoengine
https://github.com/MongoEngine/django-mongoengine

The Django one however is older and unmaintained. The mentioned lack of use as the cause in the discussion about removing the django piecese from their core.

**Comment by mgoffin** [comment]
It's also been 2 years since they've committed anything :\

**Comment by dsnellgrove** [comment]
Yep. Support for Django has been relinquished into "the community's hands" which will probably mean us if we want to continue using it.

**Comment by mgoffin** [comment]
Pretty sure the only things we'd need it for are:

https://github.com/crits/crits/blob/master/crits/settings.py#L402
https://github.com/crits/crits/blob/master/crits/settings.py#L405

It might be that we roll our own for these.

The other horrible option is that we yet again rebuild our backend ORM so that it is classful but abstract so that databases become "supportable". In this regard we can say "we support MongoDB through MongoEngine" or "We support ElasticSearch through....whatever".

**Comment by ckane** [comment]
Hmm... I'll have to look into eve and flask... I never really liked django much and thus never bothered to become proficient with it.

**Comment by webstergd** [comment]
For what it's worth, I enjoy flask. I found it to have a lot less bloat and be easier to work with then djanjo. Granted this was 2 years ago but I am sure that is all still mostly true.

**Comment by dsnellgrove** [comment]
@mgoffin I was always surprised CRITs didn't use Django's native ORM. It supports PostgreSQL, MySQL, SQLite, Oracle (officially), SAP SQL, IBM DB2, MSSQL, Firebird, ODBC and ADSDB (unofficially). Not using it seemed to leave a lot of the benefit of choosing Django on the table.

**Comment by dsnellgrove** [comment]
@mgoffin Just remembered something from a couple years ago. When CRITs was using mysql only for user auth and everything else was in mongodb I just set django to use sqlite for the user storage so I didn't have to install a second database. If there isn't a great solution for the user auth looming doing that would keep the install requirements from going up in the short term.

**Comment by mgoffin** [comment]
I'm not sure what the talk of Flask is for. The entire project is written with Django :)

The original project was written with MySQL and it scaled horribly so we moved to MongoDB (long before ever considering sharing the project with anyone).

We moved auth to MongoDB specifically to avoid multiple DB requirements for installation. I don't want to go back to that.

This is basically a non-issue until someone can figure out how to make CRITs and services work on Django > 1.7 anyways.
